File: internal/model/target.go
package model

import "time"

// Target is a piece of infrastructure that Nexus observes.
type Target struct {
	ID        string        `json:"id" db:"id"`
	Name      string        `json:"name" db:"name"`
	Provider  CloudProvider `json:"provider" db:"provider"`
	Regions   []string      `json:"regions" db:"regions"`
	Endpoint  string        `json:"endpoint" db:"endpoint"`
	Auth      TargetAuth    `json:"auth" db:"auth"`
	Status    TargetStatus  `json:"status" db:"status"`
	CreatedAt time.Time     `json:"created_at" db:"created_at"`
	UpdatedAt time.Time     `json:"updated_at" db:"updated_at"`
}

// TargetAuth describes how Nexus authenticates to a target. Credentials
// themselves never live here; only the method and non-secret hints.
type TargetAuth struct {
	Method  string `json:"method"` // env, iam, oidc, static
	Profile string `json:"profile,omitempty"`
	Region  string `json:"region,omitempty"`
}

// CloudProvider identifies the platform backing a target.
type CloudProvider string

const (
	ProviderAWS        CloudProvider = "aws"
	ProviderAzure      CloudProvider = "azure"
	ProviderGCP        CloudProvider = "gcp"
	ProviderK8s        CloudProvider = "kubernetes"
	ProviderLocalStack CloudProvider = "localstack"
)

// TargetStatus reflects connectivity health for a target.
type TargetStatus string

const (
	TargetStatusActive      TargetStatus = "active"
	TargetStatusUnreachable TargetStatus = "unreachable"
	TargetStatusDisabled    TargetStatus = "disabled"
)

File: nexus/models/target.py
from datetime import datetime
from enum import StrEnum
from uuid import uuid4

from pydantic import BaseModel, Field


class CloudProvider(StrEnum):
    aws = "aws"
    azure = "azure"
    gcp = "gcp"
    kubernetes = "kubernetes"
    localstack = "localstack"


class TargetStatus(StrEnum):
    active = "active"
    unreachable = "unreachable"
    disabled = "disabled"


class TargetAuth(BaseModel):
    method: str = "env"
    profile: str | None = None
    region: str | None = None


class Target(BaseModel):
    id: str = Field(default_factory=lambda: str(uuid4()))
    name: str
    provider: CloudProvider
    regions: list[str] = Field(default_factory=list)
    endpoint: str = ""
    auth: TargetAuth = Field(default_factory=TargetAuth)
    status: TargetStatus = TargetStatus.active
    created_at: datetime = Field(default_factory=datetime.utcnow)
    updated_at: datetime = Field(default_factory=datetime.utcnow)
